src/geometry.cpp

```diff
@@ -396,7 +400,7 @@ BoundaryInfo distance_to_boundary(Particle* p)
     // a higher level then we need to make sure that the higher level boundary
     // is selected.  This logic must consider floating point precision.
     double& d = info.distance;
-    if (d_surf < d_lat) {
+    if (d_surf < d_lat && !coincident(d_surf, d_lat)) {
```

Comment:
We can make this same check with many fewer operations by instead writing
```c++
     if (d_surf < d_lat - FP_COINCIDENT) {
```

Reply:
Good call. On it.